lib/db/schema/patch-092-093.sql

```diff
@@ -0,0 +1,47 @@
+-- This migration removes the use of `ROW_COUNT()` on the last
+-- remaining procedures.
+
+SET NAMES utf8mb4 COLLATE utf8mb4_bin;
+
+CALL assertPatchLevel('92');
+
+-- Update to not use `ROW_COUNT()`
+CREATE PROCEDURE `setPrimaryEmail_4` (
+  IN `inUid` BINARY(16),
+  IN `inNormalizedEmail` VARCHAR(255) CHARACTER SET utf8 COLLATE utf8_bin
+)
+BEGIN
+  DECLARE EXIT HANDLER FOR SQLEXCEPTION
+  BEGIN
+    ROLLBACK;
+    RESIGNAL;
+  END;
+
+  START TRANSACTION;
+     UPDATE emails SET isPrimary = false WHERE uid = inUid AND isPrimary = true;
+     UPDATE emails SET isPrimary = true WHERE uid = inUid AND isPrimary = false AND normalizedEmail = inNormalizedEmail;
+  COMMIT;
+END;
+
+-- Update to not use `ROW_COUNT()`
+CREATE PROCEDURE `consumeRecoveryCode_3` (
+  IN `uidArg` BINARY(16),
+  IN `codeHashArg` BINARY(32)
+)
+BEGIN
+  DECLARE EXIT HANDLER FOR SQLEXCEPTION
+  BEGIN
+    ROLLBACK;
+    RESIGNAL;
+  END;
+
+  START TRANSACTION;
+
+  DELETE FROM `recoveryCodes` WHERE `uid` = `uidArg` AND `codeHash` = `codeHashArg`;
+
+  SELECT COUNT(*) AS count FROM `recoveryCodes` WHERE `uid` = `uidArg`;
```

Comment:
The previous version of this procedure used `ROW_COUNT` to check whether we actually found the given code on the user's account:

```
SELECT ROW_COUNT() INTO @deletedCount;
IF @deletedCount = 0 THEN
   SIGNAL SQLSTATE '45000' SET MYSQL_ERRNO = 1643, MESSAGE_TEXT = 'Unknown recovery code.';
END IF;
```

Does the responsibility for this check now live somewhere else?

Reply:
@rfk The responsibility for this check actually lives here now, https://github.com/mozilla/fxa-auth-db-mysql/pull/445/files#diff-551d684cf1c1a288045d0e9dbce3569eL1531. I believe when we started storing scrypted values of the recovery code, this procedure wasn't updated.